osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs

```diff
@@ -503,17 +494,30 @@ public override bool OnExiting(ScreenExitEvent e)
             return base.OnExiting(e);
         }
 
-        private void applyLoopingToTrack()
+        public static void TrySetDailyChallengeBeatmap(OsuScreen screen, BeatmapManager beatmaps, RulesetStore rulesets, MusicController music, PlaylistItem item)
```

Comment:
Slight teeth pain at this method's public-static-dependency-accepting shape but I guess it's localised so probably fine. Likely better than weird inheritance anyhow